src/controllers/controller.bar.js

```diff
@@ -1,24 +1,43 @@
 import DatasetController from '../core/core.datasetController';
-import {clipArea, unclipArea} from '../helpers/helpers.canvas';
-import {isArray, isNullOrUndef, valueOrDefault, resolveObjectKey} from '../helpers/helpers.core';
-import {_limitValue, sign} from '../helpers/helpers.math';
+import {
+	clipArea, unclipArea, _arrayUnique, isArray, isNullOrUndef,
+	valueOrDefault, resolveObjectKey, _limitValue, sign
+} from '../helpers';
+
+function getAllScaleValues(scale) {
+	if (!scale._cache.$bar) {
+		const metas = scale.getMatchingVisibleMetas('bar');
+		let values = [];
+
+		for (let i = 0, ilen = metas.length; i < ilen; i++) {
+			values = values.concat(metas[i].controller.getAllParsedValues(scale));
```

Comment:
Does this need a filter if there are multiple x axes?

Reply:
No, its handled by `const metas = scale.getMatchingVisibleMetas('bar');`
This returns all dataset metas that are of type `'bar'` and attached to that scale (and visible).

Reply:
Perfect!